doc/regexp.rdoc

```diff
@@ -757,71 +1175,80 @@ performance.
 Consider a string of 25 <i>a</i>s, a <i>d</i>, 4 <i>a</i>s, and a
 <i>c</i>.
 
-    s = 'a' * 25 + 'd' + 'a' * 4 + 'c'
-    #=> "aaaaaaaaaaaaaaaaaaaaaaaaadaaaac"
+   s = 'a' * 25 + 'd' + 'a' * 4 + 'c'
+   # => "aaaaaaaaaaaaaaaaaaaaaaaaadaaaac"
 
-The following patterns match instantly as you would expect:
+The following patterns match quickly, as one would expect:
 
-    /(b|a)/ =~ s #=> 0
-    /(b|a+)/ =~ s #=> 0
-    /(b|a+)*/ =~ s #=> 0
+  (start = Time.now); /(b|a)/.match(s);   (Time.now - start) # => 4.4e-06
+  (start = Time.now); /(b|a+)/.match(s);  (Time.now - start) # => 4.6e-06
+  (start = Time.now); /(b|a+)*/.match(s); (Time.now - start) # => 5.2e-06
 
 However, the following pattern takes appreciably longer:
 
-    /(b|a+)*c/ =~ s #=> 26
+  (start = Time.now); /(b|a+)*c/.match(s); (Time.now - start) # => 5.3e-06
```

Comment:
I think there's a typo in the result, because it should take much longer than that.

Reply:
s = 'a' * 25 + 'd' + 'a' * 4 + 'c'
# => "aaaaaaaaaaaaaaaaaaaaaaaaadaaaac"
(start = Time.now); /(b|a+)*c/.match(s); (Time.now - start)
# => 7.0e-06
(start = Time.now); /(b|a+)*c/.match(s); (Time.now - start)
# => 1.07e-05
(start = Time.now); /(b|a+)*c/.match(s); (Time.now - start)
# => 7.3e-06

Is it possible that the engine has been improved?

Reply:
Yes, the engine indeed has been improved. See in particular https://bugs.ruby-lang.org/issues/19104. I think this part of the documentation should be carefully rewritten by somebody who understands the underlying issues. This should be handled separately from this pull request. @make_now_just

Reply:
Thanks, @duerst.  I agree; there's much more to some of Regexp than I understand.  I aspire only to improving its documentation, not to perfecting it.

Reply:
Thank you for mentioning it. I recently updated the Regexp optimization (see #7931). Thus, this status is a little unstable, and I will try to update the docs after stabilizing this.